### DISCUSSION TITLE
How to edit a channel/group username

### DISCUSSION THREAD

**Discussion by jiyamathias** [mymmrac/telego#81] [Q&A] [answered]
Good evening, I  am trying to build a telegram bot which can update a channel/group username (to something like t.me/groupName) but I don't seem to know which method in this package can help me achieve that.  If there is, I will be glad if I can get a sample code to better explain it.
Thanks in advance.

**Comment by mymmrac** [comment, marked as answer]
Hello, I am afraid with Telegram bots you can't do that, the only thing I can suggest is using [createChatInviteLink](https://core.telegram.org/bots/api#createchatinvitelink) and [editChatInviteLink](https://core.telegram.org/bots/api#editchatinvitelink) methods. 

Telegram does not provide a lot of things that bots can change, and channel/group username is one of them.

If you really need such functionality, you can try to look into user bots (it's basically custom Telegram client that uses real user to do some actions, like changing usernames in your case), but Telego does not provide such API (and probably will never do).

**Reply by jiyamathias** [reply]
Thank you for the response. 
You made mention of "user bots", please can you provide me with a url to help me?

**Reply by mymmrac** [reply]
Sure, here are some links you can try to research:

- https://core.telegram.org/tdlib/getting-started
- https://github.com/gotd/td
- https://github.com/Arman92/go-tdlib

**Reply by jiyamathias** [reply]
Thanks a lot.
I appreciate the time and effort taken to attend to my issue 🙌
